pymc3/distributions/bound.py

```diff
@@ -263,47 +75,61 @@ class Bound:
                     'par2', mu=0.0, sigma=1.0, initval=1.0)
     """
 
-    def __init__(self, distribution, lower=None, upper=None):
+    @classmethod
+    def dist(self, distribution, lower=None, upper=None):
         if isinstance(lower, Real) and lower == -np.inf:
             lower = None
         if isinstance(upper, Real) and upper == np.inf:
             upper = None
 
-        if not issubclass(distribution, Distribution):
-            raise ValueError('"distribution" must be a Distribution subclass.')
-
-        self.distribution = distribution
-        self.lower = lower
-        self.upper = upper
-
-    def __call__(self, name, *args, **kwargs):
-        if "observed" in kwargs:
-            raise ValueError(
-                "Observed Bound distributions are not supported. "
-                "If you want to model truncated data "
-                "you can use a pm.Potential in combination "
-                "with the cumulative probability function."
-            )
+        # if "observed" in kwargs:
```

Comment:
Maybe we can apply a `Potential` internally?

Reply:
What do you mean?

Reply:
Nevermind, I was thinking about something else.